src/IndexResolver/ProductIndexResolverInterface.php

```diff
@@ -5,16 +5,12 @@
 namespace Setono\SyliusAlgoliaPlugin\IndexResolver;
 
 use Sylius\Component\Channel\Model\ChannelInterface;
-use Sylius\Component\Locale\Model\LocaleInterface;
 
 interface ProductIndexResolverInterface
 {
     /**
      * Will return the product index name for the given channel and locale or if null is provided,
      * from the channel context and locale context respectively
-     *
-     * @param string|ChannelInterface|null $channel
-     * @param string|LocaleInterface|null $locale
      */
-    public function resolve($channel = null, $locale = null): string;
+    public function resolve(?ChannelInterface $channel = null, ?string $localeCode = null): string;
```

Comment:
The reason I didn't type hint was because it gives you better development experience. Whatever channel or locale type you have, just input it. For example the locale context returns a string. To satisfy this new signature I would then have to fetch that locale just so the resolver could get the exact same code I got in the first place from the locale context.

Reply:
But I think we should not allow loose typing.  And that's why I typed to localeCode, string.

The most frequent information we fetch are `ChannelInterface` for channel, and `string` for localeCode.

Reply:
You're right

Reply:
```suggestion
    public function resolve(ChannelInterface $channel = null, string $localeCode = null): string;
```